utils/true_up.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package utils
+
+import (
+	"fmt"
+	"time"
+)
+
+const trueUpReviewDueDay = 15
+const day = time.Hour * 24
+
+type DueDateWindow struct {
+	Start time.Time
+	End   time.Time
+}
+
+func GetNextTrueUpReviewDueDate(now time.Time) time.Time {
+	nowYear := now.Year()
+	nowMonth := now.Month()
+	nowDay := now.Day()
+	finalQuarterYear := nowYear
+	if nowMonth >= time.October && nowMonth <= time.December {
+		finalQuarterYear = nowYear + 1
+	}
+	trueUpSubmissionWindows := []DueDateWindow{
```

Comment:
2nd Quarter and 3rd Quarter should start on the 15th as their final months have only 30 days in each (June and September)

Your method looks good, but there are a few things I noticed about it. The due date should always be the 15th (15 days after the last end of the quarter), but it's often it's returning the 16th for me.
I think adding two options adds some unnecessary complexity. We really only ever need the due date from the current date and don't exactly need to look try and look for the due date outside of a window or the due date within a window.

We just need to know the next due date from today. If we're past the due date (15th) of the previous quarter, then we find the next due date automatically, since we're already past the window for the last quarter.

Reply:
Oh, I see. So your method returns the windows start date?

Or is this following `Date True Up Review is no Longer visible` I added on the table above? That column was meant to show when the panel will no longer be visible. I suppose your route makes more sense to me based on that, and the front-end would just check if it's any day before that within the previous 30 days or so.

Reply:
Gotcha, thank Allan! :)